Sources/Sentry/SentryProfiler.mm

```diff
@@ -283,20 +285,61 @@ - (instancetype)initWithHub:(SentryHub *)hub
         return nil;
     }
 
+    _profileId = [[SentryId alloc] init];
+
     SENTRY_LOG_DEBUG(@"Initialized new SentryProfiler %@", self);
     _debugImageProvider = [SentryDependencyContainer sharedInstance].debugImageProvider;
     _hub = hub;
+    [self start];
+    [self scheduleTimeoutTimer];
+
+#    if SENTRY_HAS_UIKIT
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(backgroundAbort)
+                                                 name:UIApplicationWillResignActiveNotification
+                                               object:nil];
+#    endif // SENTRY_HAS_UIKIT
+
     return self;
 }
 
+/**
+ * Schedule a timeout timer on the main thread.
+ * @warning from NSTimer.h: Timers scheduled in an async context may never fire.
+ */
+- (void)scheduleTimeoutTimer
```

Comment:
This is one of the cruxes of the change. By not ensuring we were scheduling the timeout timer from the main thread, we created a situation where the profiler may never stop, leading to unbounded memory growth.

Reply:
Any reason why we wouldn't use a dispatch timer that lets you explicitly specify which queue it fires on?

Reply:
There was no specific reason we didn't use a dispatch timer AFAIK. Do you think we should change it in this PR?

Reply:
It can be addressed separately 👍🏽